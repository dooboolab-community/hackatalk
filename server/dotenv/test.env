JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
