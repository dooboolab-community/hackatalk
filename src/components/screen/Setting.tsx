import { DefaultNavigationProps } from '../../types';
import React from 'react';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  font-size: 16;
  color: blue;
`;

interface Props {
  navigation?: DefaultNavigationProps<'Setting'>;
}

function Page(props: Props): React.ReactElement {
  return (
    <Container>
      <StyledText testID="myText">dooboolab</StyledText>
    </Container>
  );
}

export default Page;
