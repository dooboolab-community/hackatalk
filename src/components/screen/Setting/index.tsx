import {
  Container,
  HeaderContainer,
  ItemContainer,
  ItemLabel,
  SectionHeader,
} from './styles';
import React, { ReactElement } from 'react';
import { SectionList, SectionListData } from 'react-native';
import { SvgApple, SvgFacebook, SvgGoogle } from '../../../utils/Icons';

import { DefaultTheme } from 'styled-components/native';
import { FontAwesome } from '@expo/vector-icons';
import { MainStackNavigationProps } from '../../navigation/MainStackNavigator';
import { SocialType } from '../../../types';
import { getString } from '../../../../STRINGS';
import { useAuthUserContext } from '../../../providers/AuthUserProvider';
import { useThemeContext } from '@dooboo-ui/native-theme';

interface SettingsOption {
  label: string;
  icon?: ReactElement;
  onPress(): void;
  testID: string;
}

function SectionItem(
  option: SettingsOption,
  theme: DefaultTheme,
): React.ReactElement {
  return (
    <ItemContainer onPress={option.onPress} testID={option.testID}>
      {option.icon || null}
      <ItemLabel>{option.label}</ItemLabel>
      <FontAwesome name="angle-right" size={24} color={theme.fontColor} />
    </ItemContainer>
  );
}

export interface Props {
  navigation: MainStackNavigationProps;
}

function SettingScreen(props: Props): React.ReactElement {
  const { theme } = useThemeContext();
  const { navigation } = props;
  const {
    state: { user },
  } = useAuthUserContext();

  let signInInfoOption: SettingsOption;

  switch (user && user.social) {
    case SocialType.Google:
      signInInfoOption = {
        icon: <SvgGoogle width={24} fill={theme.googleIcon}/>,
        label: getString('SIGNED_IN_WITH_GOOGLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'changePwItem',
      };
      break;
    case SocialType.Facebook:
      signInInfoOption = {
        icon: <SvgFacebook width={24} fill={theme.facebookIcon}/>,
        label: getString('SIGNED_IN_WITH_FACEBOOK'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'changePwItem',
      };
      break;
    case SocialType.Apple:
      signInInfoOption = {
        icon: <SvgApple width={24} fill={theme.appleIcon}/>,
        label: getString('SIGNED_IN_WITH_APPLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'changePwItem',
      };
      break;
    case SocialType.Email:
    default:
      signInInfoOption = {
        label: getString('SIGNED_IN_WITH_EMAIL'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'changePwItem',
      };
      break;
  }

  const settings: SectionListData<SettingsOption>[] = [
    {
      title: getString('LOGIN_INFORMATION'),
      data: [signInInfoOption],
    },
  ];

  return (
    <Container>
      <SectionList
        testID="mySectionList"
        sections={settings}
        renderItem={({ item }): React.ReactElement => SectionItem(item, theme)}
        keyExtractor={(item: SettingsOption): string => item.label}
        renderSectionHeader={({ section: { title } }): React.ReactElement => (
          <HeaderContainer>
            <SectionHeader>{title}</SectionHeader>
          </HeaderContainer>
        )}
      />
    </Container>
  );
}

export default SettingScreen;
