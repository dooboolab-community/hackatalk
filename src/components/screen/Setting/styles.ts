import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  padding-top: 10px;
  background-color: ${({ theme }): string => theme.background};
`;

export const HeaderContainer = styled.View`
  background-color: ${({ theme }): string => theme.background};
  height: 40px;
  justify-content: center;
  margin-left: 12px;
  border-bottom-width: 1;
  border-bottom-color: ${({ theme }): string => theme.lineColor};
`;

export const SectionHeader = styled.Text`
  color: ${({ theme }): string => theme.fontSubColor};
  margin-left: 2px;
`;

export const ItemContainer = styled.TouchableOpacity`
  flex-direction: row;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: 52px;
  align-items: center;
  /* border-width: 1px; */
`;

export const ItemIcon = styled.Image`
  width: 32px;
  height: 32px;
  border-width: 1;
  border-color: ${({ theme }): string => theme.lineColor};
  border-radius: 4;
  margin-right: 10px;
`;

export const ItemLabel = styled.Text`
  color: ${({ theme }): string => theme.fontColor};
  font-size: 16px;
  flex: 1;
`;
