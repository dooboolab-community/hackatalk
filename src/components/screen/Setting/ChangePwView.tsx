import { Alert, KeyboardAvoidingView, SafeAreaView } from 'react-native';
import {
  DefaultTheme, withTheme,
} from 'styled-components/native';
import { InnerContainer, ModalCloseButton, ModalHeader, ModalTitle, StyledTextInput } from './styles';
import React, { useState } from 'react';

import Button from '../../shared/Button';
import { Ionicons } from '@expo/vector-icons';
import { getString } from '../../../../STRINGS';

export interface Props {
    theme: DefaultTheme;
    close(): void;
    checkCurrentPw(): Promise<boolean>;
}

function ChangePwView(props: Props): React.ReactElement {
  const { theme, close, checkCurrentPw } = props;
  const [currentPwVerified, setCurrentPwVerified] = useState(false);
  const [currentPw, setCurrentPw] = useState('');
  const [newPw, setNewPw] = useState('');
  const [newPwCheck, setNewPwCheck] = useState('');
  const checkCurrent = (): void => {
    checkCurrentPw().then((verified) => {
      if (verified) {
        setCurrentPwVerified(true);
      } else {
        throw Error(getString('PASSWORD_MUST_MATCH'));
      }
    }).catch((err: Error) => {
      Alert.alert('', err.message);
    });
  };
  const changePassword = async (): Promise<void> => {
    if (newPw === newPwCheck) {
      // TODO change password api call
      Alert.alert('', 'Password changed.', [
        {
          text: getString('OK'),
          onPress: (): void => {
            close();
          },
        },
      ]);
    } else {
      Alert.alert(getString('PASSWORD_MUST_MATCH'));
    }
  };
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <KeyboardAvoidingView style={{ flex: 1 }} behavior="padding">
        <ModalHeader>
          <ModalTitle>{getString('PASSWORD_CHANGE')}</ModalTitle>
          <ModalCloseButton
            testID="closeBtn"
            onPress={(): void | null => close()}>
            <Ionicons name="md-close" size={32} color={theme.fontColor}/>
          </ModalCloseButton>
        </ModalHeader>
        {
          currentPwVerified
            ? <InnerContainer>
              <StyledTextInput
                key="newPwTextInput"
                testID="newPwTextInput"
                isPassword
                onTextChanged={(pw): void => setNewPw(pw)}
                txtLabel={getString('PASSWORD_NEW')}/>
              <StyledTextInput
                key="newPwCheckTextInput"
                testID="newPwCheckTextInput"
                isPassword
                onTextChanged={(pw): void => setNewPwCheck(pw)}
                txtLabel={getString('PASSWORD_NEW_CHECK')}/>
            </InnerContainer>
            : <InnerContainer>
              <StyledTextInput
                key="currentPwTextInput"
                testID="currentPwTextInput"
                isPassword
                onTextChanged={(pw): void => setCurrentPw(pw)}
                txtLabel={getString('PASSWORD_CURRENT')}/>
            </InnerContainer>
        }
        <Button
          testID="checkCurrentPwBtn"
          onPress={currentPwVerified ? changePassword : checkCurrent}>
          {currentPwVerified ? getString('CONFIRM') : getString('NEXT')}
        </Button>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

export default withTheme(ChangePwView);
