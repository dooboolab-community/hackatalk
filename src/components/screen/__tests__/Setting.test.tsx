import 'react-native';

import * as React from 'react';

import { RenderResult, render } from '../../../../test/test-utils';

import Setting from '../Setting';
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object): object => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[Setting] screen', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = <Setting {...props} />;
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('should render [Text] with value "myText"', () => {
    const textInstance = testingLib.getByTestId('myText');
    expect(textInstance.props.children).toEqual('dooboolab');
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onPress', () => {
      // const btn = testingLib.queryByTestId('btn');
      // act(() => {
      //   fireEvent.press(btn);
      //   fireEvent.press(btn);
      // });
      // expect(cnt).toBe(3);
    });
  });
});
