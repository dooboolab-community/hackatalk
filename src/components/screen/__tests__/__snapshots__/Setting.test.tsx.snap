// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Setting] screen renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <ScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "label": "Signed in with Email",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    stickySectionHeadersEnabled={true}
    testID="mySectionList"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#d3d8e8",
              "borderBottomWidth": 1,
              "height": 52,
              "justifyContent": "center",
              "marginLeft": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1E6EFA",
              },
            ]
          }
        >
          Login information
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            },
          ]
        }
        testID="changePwItem"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#151A25",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
              },
            ]
          }
        >
          Signed in with Email
        </Text>
        <Text />
      </TouchableOpacity>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </ScrollView>
</View>
`;

exports[`[Setting] screen renders without crashing 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <ScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "icon": <SvgMock
                fill="#ffffff"
                width={24}
              />,
              "label": "Signed in with Facebook",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "icon": <SvgMock
                fill="#ffffff"
                width={24}
              />,
              "label": "Signed in with Facebook",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    stickySectionHeadersEnabled={true}
    testID="mySectionList"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#d3d8e8",
              "borderBottomWidth": 1,
              "height": 52,
              "justifyContent": "center",
              "marginLeft": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1E6EFA",
              },
            ]
          }
        >
          Login information
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            },
          ]
        }
        testID="changePwItem"
      >
        <SvgMock
          fill="#ffffff"
          width={24}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#151A25",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
              },
            ]
          }
        >
          Signed in with Facebook
        </Text>
        <Text />
      </TouchableOpacity>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </ScrollView>
</View>
`;

exports[`[Setting] screen renders without crashing 3`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingTop": 10,
      },
    ]
  }
>
  <ScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "icon": <SvgMock
                fill="#ffffff"
                width={24}
              />,
              "label": "Signed in with Google",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "icon": <SvgMock
                fill="#ffffff"
                width={24}
              />,
              "label": "Signed in with Google",
              "onPress": [Function],
              "testID": "changePwItem",
            },
          ],
          "title": "Login information",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    stickySectionHeadersEnabled={true}
    testID="mySectionList"
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#d3d8e8",
              "borderBottomWidth": 1,
              "height": 52,
              "justifyContent": "center",
              "marginLeft": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1E6EFA",
              },
            ]
          }
        >
          Login information
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 52,
              "paddingLeft": 15,
              "paddingRight": 15,
              "width": "100%",
            },
          ]
        }
        testID="changePwItem"
      >
        <SvgMock
          fill="#ffffff"
          width={24}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#151A25",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
              },
            ]
          }
        >
          Signed in with Google
        </Text>
        <Text />
      </TouchableOpacity>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </ScrollView>
</View>
`;
